### PR TITLE
feat: add grpc requests for invite service

### PR DESCRIPTION
**Pull request by skelet0no**
1. add messages in proto
2. add get invite by invite id method

### CODE REVIEW

File: invite_service/src/invite_service_impl.py [hunk 1]
```diff
@@ -84,9 +108,64 @@ def get_all_invites(
...
+        Returns
+        -------
+        proto.InviteResponse
+            Response object for invite response.
+
+        """
+        try:
+            invite = await self._invite_repository.get_invite_by_invite_id(
+                invite_id=request.invite_id
+            )
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:178]: check if user is allowed to get the invite

File: invite_service/src/invite_service_impl.py [hunk 2]
```diff
@@ -105,10 +184,28 @@ def get_invites_by_invitee_id(
             Invites object for invite response.
 
         """
-        pass
-
-    def create_invite(
-        self, request: proto.Invite, context: grpc.ServicerContext
+        try:
+            invites = await self._invite_repository.get_invites_by_invitee_id(
+                invitee_id=request.invite_id,
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py]: maybe invitee_id

File: invite_service/src/invite_service_impl.py [hunk 3]
```diff
@@ -147,9 +254,19 @@ def update_invite(
             Object containing status code and message if the response status is not 200.
 
         """
-        pass
-
-    def delete_invite(
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:312]: check if user can modify invite

File: invite_service/src/invite_service_impl.py [hunk 4]
```diff
@@ -168,4 +285,13 @@ def delete_invite(
             Object containing status code and message if the response status is not 200.
 
         """
-        pass
+        try:
+            await self._invite_repository.delete_invite(invite_id=request.invite_id)
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:356]: check if user can delete invite

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

File: invite_service/proto/invite_service.proto [hunk 5]
```diff
@@ -1,21 +1,40 @@
...
+
+message InviteRequestByInviteId {
+  string invite_id = 1;
+  GrpcInvite invite = 2;
+  GrpcUser user = 3;
+}
+
+message InviteResponse {
+  int32 status_code = 1;
+  GrpcInvite invite = 2;
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: must be optional

File: invite_service/proto/invite_service.proto [hunk 6]
```diff
@@ -1,21 +1,40 @@
...
+
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/empty.proto";
 
 service InviteService {
+  rpc get_invite_by_invite_id(InviteRequestByInviteId) returns (InviteResponse){}
   rpc get_all_invites(google.protobuf.Empty) returns (InvitesResponse){}
-  rpc get_invites_by_author_id(InvitesByUserIdRequest) returns (InvitesResponse){}
+  rpc get_invites_by_author_id(InvitesByAuthorIdRequest) returns (InvitesResponse){}
   rpc get_invites_by_invitee_id(GetInviteeByInviteIdRequest) returns (InvitesResponse){}
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: must be GetInvitesByInviteeIdRequest

File: invite_service/proto/invite_service.proto [hunk 7]
```diff
@@ -1,21 +1,40 @@
 syntax = "proto3";
+
 import "google/protobuf/timestamp.proto";
 import "google/protobuf/empty.proto";
 
 service InviteService {
+  rpc get_invite_by_invite_id(InviteRequestByInviteId) returns (InviteResponse){}
   rpc get_all_invites(google.protobuf.Empty) returns (InvitesResponse){}
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: you need to throw user in request to check user type, add new message RequestingUser

File: invite_service/repository/invite_repository_impl.py [hunk 8]
```diff
@@ -78,6 +78,35 @@ async def get_invites_by_author_id(self, author_id: str) -> List[Invite]:
             for db_invite in db_invites
         ]
 
+    async def get_invite_by_invite_id(self, invite_id: str) -> Invite:
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py:84]: add this method to class methods docstring

File: invite_service/src/invite_service_impl.py [hunk 9]
```diff
@@ -126,10 +229,20 @@ def create_invite(
             Object containing status code and message if the response status is not 200.
 
         """
-        pass
-
-    def update_invite(
-        self, request: proto.Invite, context: grpc.ServicerContext
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
+            await self._invite_repository.create_invite(invite=invite)
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:282]: check if user can create invite

**Review by ZixMai — CHANGES_REQUESTED** [review]
CHECK FOR ADMIN USER TYPE, DONT RESOLVE CONVERSATION IF YOU DONT UNDERSTAND PROBLEM

File: invite_service/src/invite_service_impl.py [hunk 10]
```diff
@@ -58,9 +64,27 @@ def get_invites_by_author_id(
...
+            context.set_code(grpc.StatusCode.NOT_FOUND)
+            return proto.InvitesResponse(status_code=404, message="Invite not found")
+        except prisma.errors.PrismaError:
+            context.set_code(grpc.StatusCode.INTERNAL)
+            return proto.InvitesResponse(
+                status_code=500, message="Internal server error"
+            )
+
+    async def get_all_invites(
         self, request: Empty, context: grpc.ServicerContext
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: add requesting user from proto

File: invite_service/src/invite_service_impl.py [hunk 11]
```diff
@@ -84,9 +108,70 @@ def get_all_invites(
             Response object for invite response.
 
         """
-        pass
+        try:
+            if request.user.type != proto.GrpcUserType.ADMIN:
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: WHERE DID YOU FIND USER TYPE IN EMPTY OBJECT

**Review by ZixMai — CHANGES_REQUESTED** [review]

File: invite_service/proto/invite_service.proto [hunk 12]
```diff
@@ -1,21 +1,39 @@
...
 
-message InvitesByUserIdRequest {
-  string user_id = 1;
+message InvitesByAuthorIdRequest {
+  string author_id = 1;
+}
+
+message InviteRequestByInviteId {
+  string invite_id = 1;
+  GrpcInvite invite = 2;
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: remove this field

File: invite_service/proto/invite_service.proto [hunk 13]
```diff
@@ -1,21 +1,39 @@
...
-  rpc get_all_invites(google.protobuf.Empty) returns (InvitesResponse){}
-  rpc get_invites_by_author_id(InvitesByUserIdRequest) returns (InvitesResponse){}
-  rpc get_invites_by_invitee_id(GetInviteeByInviteIdRequest) returns (InvitesResponse){}
-  rpc create_invite(Invite) returns (BaseResponse){}
-  rpc update_invite(Invite) returns (BaseResponse){}
+  rpc get_invite_by_invite_id(InviteRequestByInviteId) returns (InviteResponse){}
+  rpc get_all_invites(RequestingUser) returns (InvitesResponse){}
+  rpc get_invites_by_author_id(InvitesByAuthorIdRequest) returns (InvitesResponse){}
+  rpc get_invites_by_invitee_id(GetInvitesByInviteeIdRequest) returns (InvitesResponse){}
+  rpc create_invite(GrpcInvite) returns (BaseResponse){}
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: must be InviteRequest instead of GrpcInvite

File: invite_service/proto/invite_service.proto [hunk 14]
```diff
@@ -1,21 +1,39 @@
...
+  rpc update_invite(GrpcInvite) returns (BaseResponse){}
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: must be InviteRequest instead of GrpcInvite

File: invite_service/proto/invite_service.proto [hunk 15]
```diff
@@ -25,31 +43,51 @@ message Invite {
...
   string invite_id = 1;
+  GrpcInvite invite = 2;
+  GrpcUser user = 3;
 }
 
 message GetInvitesByInviteeIdRequest {
   string invitee_id = 1;
 }
 
 message GetInviteeByInviteIdRequest {
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: remove this message

File: invite_service/proto/invite_service.proto [hunk 16]
```diff
@@ -25,31 +43,51 @@ message Invite {
...
 
 message BaseResponse {
-  int64 code = 1;
+  int32 status_code = 1;
   optional string message = 2;
 }
 
 message DeleteInviteRequest {
   string invite_id = 1;
+  GrpcInvite invite = 2;
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: remove this field

File: invite_service/repository/invite_repository_impl.py [hunk 17]
```diff
@@ -136,7 +165,7 @@ async def get_invites_by_invitee_id(self, invitee_id: str) -> List[Invite]:
             }
         )
         if db_invites is None or len(db_invites) == 0:
-            raise ValueNotFoundError("Events not found")
+            raise ValueNotFoundError("Invites not found")
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py:171]: change was to were in docstring (not only in this method, check other methods too)

File: invite_service/repository/invite_repository_interface.py [hunk 18]
```diff
@@ -51,6 +53,31 @@ async def get_invites_by_author_id(self, author_id: str) -> List[Invite]:
...
+        -------
+        Invite
+            Invite that has matches with given invite id.
+
+        Raises
+        ------
+        prisma.errors.PrismaError
+            Catch all for every exception raised by Prisma Client Python.
+        ValueNotFoundError
+            No invites was found for given author id.
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_interface.py]: invites were*, check for this mistake in other docstrings

File: invite_service/proto/invite_service.proto [hunk 19]
```diff
@@ -1,21 +1,39 @@
...
+  rpc get_invites_by_invitee_id(GetInvitesByInviteeIdRequest) returns (InvitesResponse){}
+  rpc create_invite(GrpcInvite) returns (BaseResponse){}
+  rpc update_invite(GrpcInvite) returns (BaseResponse){}
   rpc delete_invite(DeleteInviteRequest) returns (BaseResponse){}
 }
 
-message InvitesByUserIdRequest {
-  string user_id = 1;
+message InvitesByAuthorIdRequest {
+  string author_id = 1;
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: add user field

File: invite_service/proto/invite_service.proto [hunk 20]
```diff
@@ -25,31 +43,51 @@ message Invite {
...
 }
 
 message DeleteInviteRequest {
   string invite_id = 1;
+  GrpcInvite invite = 2;
+  GrpcUser user = 3;
 }
 
 message GetInvitesByInviteeIdRequest {
   string invitee_id = 1;
```
Comment (ZixMai) [inline invite_service/proto/invite_service.proto]: add user field

File: invite_service/src/invite_service_impl.py [hunk 21]
```diff
@@ -58,17 +64,35 @@ def get_invites_by_author_id(
             Invites object for invite response.
 
         """
-        pass
-
-    def get_all_invites(
-        self, request: Empty, context: grpc.ServicerContext
+        try:
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:76]: add full user type check

File: invite_service/src/invite_service_impl.py [hunk 22]
```diff
@@ -105,10 +184,28 @@ def get_invites_by_invitee_id(
             Invites object for invite response.
 
         """
-        pass
-
-    def create_invite(
-        self, request: proto.Invite, context: grpc.ServicerContext
+        try:
+            invites = await self._invite_repository.get_invites_by_invitee_id(
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:229]: add full user type check

File: invite_service/src/invite_service_impl.py [hunk 23]
```diff
@@ -84,9 +108,64 @@ def get_all_invites(
             Response object for invite response.
 
         """
-        pass
+        try:
+            if request.user.type != proto.GrpcUserType.ADMIN:
+                raise PermissionDeniedError("Permission denied")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:130]: add raises to docstring

File: invite_service/src/invite_service_impl.py [hunk 24]
```diff
@@ -84,9 +108,64 @@ def get_all_invites(
...
+        Returns
+        -------
+        proto.InviteResponse
+            Response object for invite response.
+
+        """
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
+            if request.user.id != invite.author_id:
+                raise PermissionDeniedError("Permission denied")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: add raises to docstring

File: invite_service/src/invite_service_impl.py [hunk 25]
```diff
@@ -126,10 +223,25 @@ def create_invite(
...
-
-    def update_invite(
-        self, request: proto.Invite, context: grpc.ServicerContext
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
+            if (
+                request.user.type != proto.GrpcUserType.ADMIN
+                and request.user.id != invite.author_id
+            ):
+                raise PermissionDeniedError("Permission denied")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:281]: add raises to docstring

File: invite_service/src/invite_service_impl.py [hunk 26]
```diff
@@ -147,9 +259,22 @@ def update_invite(
...
 
         """
-        pass
-
-    def delete_invite(
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
+            if request.user.id != invite.author_id:
+                context.set_code(grpc.StatusCode.PERMISSION_DENIED)
+                return proto.BaseResponse(status_code=403, message="Permission denied")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: use PermissionDeniedError instead

File: invite_service/src/invite_service_impl.py [hunk 27]
```diff
@@ -168,4 +293,16 @@ def delete_invite(
             Object containing status code and message if the response status is not 200.
 
         """
-        pass
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
+            if request.user.id != invite.author_id:
+                raise PermissionDeniedError("Permission denied")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:355]: add raises to docstring

File: invite_service/src/invite_service_impl.py [hunk 28]
```diff
@@ -84,9 +108,64 @@ def get_all_invites(
...
+            Request context.
+
+        Returns
+        -------
+        proto.InviteResponse
+            Response object for invite response.
+
+        """
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: remove this line

File: invite_service/src/invite_service_impl.py [hunk 29]
```diff
@@ -84,9 +108,64 @@ def get_all_invites(
...
+            if request.user.id != invite.author_id:
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: move this code block under invite getting

**Review by ZixMai — CHANGES_REQUESTED** [review]
Dont resolve any conversations again... And read old problem descriptions. add admin user type check

File: invite_service/src/invite_service_impl.py [hunk 30]
```diff
@@ -168,4 +293,16 @@ def delete_invite(
             Object containing status code and message if the response status is not 200.
 
         """
-        pass
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: must be request by invite id to db instead

**Review by ZixMai — CHANGES_REQUESTED** [review]

File: invite_service/proto/invite_service.proto [hunk 31]
```diff
@@ -1,21 +1,39 @@
...
+  rpc create_invite(InviteRequest) returns (BaseResponse){}
+  rpc update_invite(InviteRequest) returns (BaseResponse){}
   rpc delete_invite(DeleteInviteRequest) returns (BaseResponse){}
 }
 
-message InvitesByUserIdRequest {
-  string user_id = 1;
+message InvitesByAuthorIdRequest {
+  string author_id = 1;
+  GrpcUser user = 2;
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto]: rename to requesting_user

File: invite_service/proto/invite_service.proto (hunk 19 above)
Comment (Nhsdkk) [reply invite_service/proto/invite_service.proto]: fixed by him

File: invite_service/proto/invite_service.proto [hunk 32]
```diff
@@ -1,21 +1,39 @@
...
+}
+
+message InviteRequestByInviteId {
+  string invite_id = 1;
+  GrpcUser user = 2;
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto]: rename to requesting_user

File: invite_service/proto/invite_service.proto [hunk 33]
```diff
@@ -1,21 +1,39 @@
...
+message InviteResponse {
+  int32 status_code = 1;
+  optional GrpcInvite invite = 2;
+  optional string message = 3;
 }
 
-message Invite {
+message InviteRequest {
+  GrpcInvite invite = 1;
+  GrpcUser user = 2;
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto]: rename to requesting_user

File: invite_service/proto/invite_service.proto (hunk 20 above)
Comment (Nhsdkk) [reply invite_service/proto/invite_service.proto]: fixed by him

File: invite_service/proto/invite_service.proto [hunk 34]
```diff
@@ -25,31 +43,47 @@ message Invite {
...
 }
 
 message DeleteInviteRequest {
   string invite_id = 1;
+  GrpcUser user = 2;
 }
 
 message GetInvitesByInviteeIdRequest {
   string invitee_id = 1;
+  GrpcUser user = 2;
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto]: rename to requesting_user

File: invite_service/proto/invite_service.proto [hunk 35]
```diff
@@ -25,31 +43,47 @@ message Invite {
...
+  string id = 1;
+  string username = 2;
+  string email = 3;
+  google.protobuf.Timestamp created_at = 4;
+  optional google.protobuf.Timestamp suspended_at = 5;
+  GrpcUserType type = 6;
+}
+
+message RequestingUser {
+  GrpcUser user = 1;
```
Comment (Nhsdkk) [inline invite_service/proto/invite_service.proto]: rename to requesting_user

File: invite_service/repository/invite_repository_impl.py (hunk 8 above)
Comment (Nhsdkk) [reply invite_service/repository/invite_repository_impl.py:84]: fixed by him

File: invite_service/repository/invite_repository_impl.py [hunk 36]
```diff
@@ -124,7 +155,7 @@ async def get_invites_by_invitee_id(self, invitee_id: str) -> List[Invite]:
         prisma.errors.PrismaError
             Catch all for every exception raised by Prisma Client Python.
         ValueNotFoundError
-            No invites was found for given invitee id.
+            No invites were found for given invitee id.
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py:159]: Great, now switch 146 line invitee id object to Invitee's id :D

**Comment by Nhsdkk** [comment]
line 165 in invite_repository_impl.py should be "invitee_id": invitee_id instead of "id": invitee_id

File: invite_service/repository/invite_repository_impl.py (hunk 17 above)
Comment (Nhsdkk) [reply invite_service/repository/invite_repository_impl.py:171]: fixed by him

File: invite_service/src/invite_service_impl.py [hunk 37]
```diff
@@ -14,23 +19,24 @@ class InviteServiceImpl(GrpcServicer):
 
     Attributes
     ----------
-    invites : List[Invite]
-        List of invites.
+    _invite_repository: InviteRepositoryInterface
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:25]: No docstring here

File: invite_service/src/invite_service_impl.py (hunk 21 above)
Comment (Nhsdkk) [reply invite_service/src/invite_service_impl.py:76]: fixed by him

File: invite_service/src/invite_service_impl.py [hunk 38]
```diff
@@ -82,19 +113,80 @@ def get_all_invites(
...
+        PermissionDeniedError
+            Raises when user dont has enough access.
+
+        """
+        try:
+            invite = await self._invite_repository.get_invite_by_invite_id(
+                invite_id=request.invite_id
+            )
+            if (
+                request.user.id != invite.author_id
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:166]: invitee can also access this

File: invite_service/src/invite_service_impl.py (hunk 22 above)
Comment (Nhsdkk) [reply invite_service/src/invite_service_impl.py:229]: fixed by him

File: invite_service/src/invite_service_impl.py [hunk 39]
```diff
@@ -124,12 +241,29 @@ def create_invite(
...
+            Raises when user don't have enough access.
 
         """
-        pass
-
-    def update_invite(
-        self, request: proto.Invite, context: grpc.ServicerContext
+        try:
+            invite = Invite.from_grpc_invite(request.invite)
+            if (
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:277]: Make this function accessible to anyone

File: invite_service/src/invite_service_impl.py (hunk 3 above)
Comment (Nhsdkk) [reply invite_service/src/invite_service_impl.py:312]: fixed

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

File: invite_service/src/invite_service_impl.py (hunk 4 above)
Comment (Nhsdkk) [reply invite_service/src/invite_service_impl.py:356]: fixed

File: invite_service/repository/invite_repository_interface.py [hunk 40]
```diff
@@ -91,7 +118,7 @@ async def get_invites_by_invitee_id(self, invitee_id: str) -> List[Invite]:
         prisma.errors.PrismaError
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_interface.py:119]: change 109'th line to Invitee's id.

**Review by ZixMai — CHANGES_REQUESTED** [review]

File: invite_service/src/invite_service_impl.py [hunk 41]
```diff
@@ -145,11 +279,28 @@ def update_invite(
         -------
         proto.BaseResponse
             Object containing status code and message if the response status is not 200.
+        PermissionDeniedError
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: Must be a raises block

**Review by ZixMai — CHANGES_REQUESTED** [review]

File: invite_service/repository/invite_repository_interface.py [hunk 42]
```diff
@@ -13,6 +14,8 @@ class InviteRepositoryInterface(ABC):
     -------
     async get_invites_by_author_id(author_id)
         Returns invites that has matches with given author id.
+    async get_invite_by_invite_id(invitee_id)
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_interface.py]: change argument to invite id

File: invite_service/repository/invite_repository_impl.py [hunk 43]
```diff
@@ -63,13 +66,13 @@ async def get_invites_by_author_id(self, author_id: str) -> List[Invite]:
...
+            No invites were found for given author id.
 
         """
-        db_invites: Optional[
-            List[PrismaInvite]
-        ] = await self._db_client.db.invite.find_many(
-            where={"id": author_id, "deleted_at": None}
+        db_invites: Optional[List[PrismaInvite]] = (
+            await self._db_client.db.invite.find_many(
+                where={"id": author_id, "deleted_at": None}
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py]: change "id" field to "author_id"

File: invite_service/src/invite_service_impl.py [hunk 44]
```diff
@@ -39,8 +47,8 @@ class InviteServiceImpl(GrpcServicer):
     def __init__(self, invite_repository: InviteRepositoryInterface) -> None:
         self._invite_repository = invite_repository
 
-    def get_invites_by_author_id(
-        self, request: proto.InvitesByUserIdRequest, context: grpc.ServicerContext
+    async def get_invites_by_author_id(
+        self, request: proto.InvitesByAuthorIdRequest, context: grpc.ServicerContext
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:53]: change parameter in docstring

File: invite_service/src/invite_service_impl.py [hunk 45]
```diff
@@ -104,11 +209,41 @@ def get_invites_by_invitee_id(
...
+            context.set_code(grpc.StatusCode.NOT_FOUND)
+            return proto.InvitesResponse(status_code=404, message="Invites not found")
+        except prisma.errors.PrismaError:
+            context.set_code(grpc.StatusCode.INTERNAL)
+            return proto.InvitesResponse(
+                status_code=500, message="Internal server error"
+            )
+
+    async def create_invite(
+        self, request: proto.InviteRequest, context: grpc.ServicerContext
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:252]: change parameter in docstring

File: invite_service/src/invite_service_impl.py [hunk 46]
```diff
@@ -125,11 +260,33 @@ def create_invite(
...
+            return proto.BaseResponse(status_code=200)
+        except ValueNotFoundError:
+            context.set_code(grpc.StatusCode.NOT_FOUND)
+            return proto.BaseResponse(status_code=404, message="Invite not found")
+        except prisma.errors.PrismaError:
+            context.set_code(grpc.StatusCode.INTERNAL)
+            return proto.BaseResponse(status_code=500, message="Internal server error")
+
+    async def update_invite(
+        self, request: proto.InviteRequest, context: grpc.ServicerContext
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:293]: change parameter in docstring

**Review by ZixMai — CHANGES_REQUESTED** [review]
5 more...

**Review by ZixMai — APPROVED** [review]

**Review by MaxPlays35 — APPROVED** [review]

**Comment by Nhsdkk** [comment]
Please merge main into this branch and make changes for imports

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Review by ZixMai — APPROVED** [review]

File: invite_service/repository/invite_repository_impl.py [hunk 47]
```diff
@@ -79,6 +81,35 @@ async def get_invites_by_author_id(self, author_id: str) -> List[Invite]:
...
+            Catch all for every exception raised by Prisma Client Python.
+        ValueNotFoundError
+            No invite was found for given invite id.
+
+        """
+        db_invite: Optional[PrismaInvite] = await self._db_client.db.invite.find_first(
+            where={"id": invite_id, "deleted_at": None}
+        )
+        if db_invite is None:
+            raise ValueNotFoundError("Invitee's not found")
```
Comment (ZixMai) [inline invite_service/repository/invite_repository_impl.py]: must be invite not found

File: invite_service/src/invite_service_impl.py [hunk 48]
```diff
@@ -58,44 +67,150 @@ def get_invites_by_author_id(
...
+            context.set_code(grpc.StatusCode.OK)
+            return proto.InvitesResponse(
+                status_code=200,
+                invites=proto.ListOfInvites(
+                    invites=[invite.to_grpc_invite() for invite in invites]
+                ),
+            )
+        except ValueNotFoundError:
+            context.set_code(grpc.StatusCode.NOT_FOUND)
+            return proto.InvitesResponse(status_code=404, message="Invite not found")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: must be invites not found

File: invite_service/src/invite_service_impl.py [hunk 49]
```diff
@@ -58,44 +67,150 @@ def get_invites_by_author_id(
...
+                request.requesting_user.id != invite.author_id
+                and request.requesting_user.id != invite.invitee_id
+                and request.requesting_user.type != GrpcUserType.ADMIN
+            ):
+                raise PermissionDeniedError("Permission denied")
+            context.set_code(grpc.StatusCode.OK)
+            return proto.InviteResponse(status_code=200, invite=invite.to_grpc_invite())
+        except ValueNotFoundError:
+            context.set_code(grpc.StatusCode.NOT_FOUND)
+            return proto.InviteResponse(status_code=404, message="Invites not found")
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: must be invite not found

File: invite_service/src/invite_service_impl.py [hunk 50]
```diff
@@ -126,18 +274,43 @@ def create_invite(
...
+            invite = Invite.from_grpc_invite(request.invite)
+            if (
+                request.requesting_user.id != invite.author_id
+                and request.requesting_user.type != GrpcUserType.ADMIN
+            ):
+                raise PermissionDeniedError("Permission denied")
+            await self._invite_repository.create_invite(invite=invite)
+            context.set_code(grpc.StatusCode.OK)
+            return proto.BaseResponse(status_code=200)
+        except ValueNotFoundError:
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: remove this exception

File: invite_service/src/invite_service_impl.py [hunk 51]
```diff
@@ -58,44 +67,150 @@ def get_invites_by_author_id(
         proto.InvitesResponse
             Invites object for invite response.
 
-        """
-        pass
+        Raises
+        ------
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:71]: remove prisma.errors from raises

File: invite_service/src/invite_service_impl.py [hunk 52]
```diff
@@ -58,44 +67,150 @@ def get_invites_by_author_id(
...
 
         Returns
         -------
-        List[Invite]
-            List of all invites.
+        proto.InvitesResponse
+            Response object for invite response.
 
         Raises
         ------
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:123]: remove prisma.errors from raises

File: invite_service/src/invite_service_impl.py [hunk 53]
```diff
@@ -58,44 +67,150 @@ def get_invites_by_author_id(
...
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        proto.InviteResponse
+            Response object for invite response.
+
+        Raises
+        ------
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:170]: remove prisma.errors from raises

File: invite_service/src/invite_service_impl.py [hunk 54]
```diff
@@ -105,18 +220,51 @@ def get_invites_by_invitee_id(
         proto.InvitesResponse
             Invites object for invite response.
 
-        """
-        pass
+        Raises
+        ------
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:218]: remove prisma.errors from raises

File: invite_service/src/invite_service_impl.py [hunk 55]
```diff
@@ -147,10 +320,25 @@ def update_invite(
         proto.BaseResponse
             Object containing status code and message if the response status is not 200.
 
-        """
-        pass
+        Raises
+        ------
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: remove prisma.errors from raises

File: invite_service/src/invite_service_impl.py [hunk 56]
```diff
@@ -168,5 +356,32 @@ def delete_invite(
         proto.BaseResponse
             Object containing status code and message if the response status is not 200.
 
+        Raises
+        ------
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py:342]: remove prisma.errors from raises

**Review by ZixMai — APPROVED** [review]

**Review by ZixMai — CHANGES_REQUESTED** [review]
waiting for fixed mistakes

File: invite_service/repository/mock_invite_repository.py
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:1]: Annotate class with singleton

**Review by ZixMai — CHANGES_REQUESTED** [review]

File: invite_service/src/invite_service_impl.py [hunk 57]
```diff
@@ -126,18 +274,43 @@ def create_invite(
         proto.BaseResponse
             Object containing status code and message if the response status is not 200.
 
-        """
-        pass
+        Raises
+        ------
+        prisma.errors.PrismaError
```
Comment (ZixMai) [inline invite_service/src/invite_service_impl.py]: remove prisma.errors from raises

**Review by ZixMai — CHANGES_REQUESTED** [review]

**Review by ZixMai — APPROVED** [review]

**Review by ZixMai — APPROVED** [review]

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**